rclcpp/resource/logging.hpp.em

```diff
@@ -45,13 +45,20 @@
 #endif
 
 @{
+from collections import OrderedDict
 from rcutils.logging import feature_combinations
 from rcutils.logging import get_macro_parameters
 from rcutils.logging import get_suffix_from_features
 from rcutils.logging import severities
+from rcutils.logging import throttle_args
+from rcutils.logging import throttle_params
 
-# TODO(dhood): Implement the throttle macro using time sources available in rclcpp
-excluded_features = ['named', 'throttle']
+throttle_args['condition_before'] = 'RCUTILS_LOG_CONDITION_THROTTLE_BEFORE(clock, duration)'
+del throttle_params['get_time_point_value']
+throttle_params['clock'] = 'rclcpp::Clock that will be used to get the time point.'
+throttle_params.move_to_end('clock', last=False)
```

Comment:
It is unfortunate that this requires changing the global state in `rcutils`. It would be good to refactor this in the future that the same can be done without altering it (e.g. using function to generate the state on demand).